### PR TITLE
adds TOC to the spec.md file

### PR DESCRIPTION
**Pull request by andorsk**
adds toc to the spec.md file. Uses markdown generator from https://github.com/ardumont/markdown-toc

### CODE REVIEW

File: spec.md [hunk 1]
```diff
@@ -42,6 +42,57 @@
 
 *Copyright: 2022 Trust Over IP Foundation*
 
+**TOC**
+<!-- markdown-toc start - Don't edit this section. Run M-x markdown-toc-refresh-toc -->
+
+**Table of Contents**
+
+- [Trust over IP (ToIP) Technology Architecture Specification](#trust-over-ip-toip-technology-architecture-specification)
```
Comment (talltree) [inline spec.md:49]: Could we remove this line, since that heading is the title of the spec and not a section inside it?

File: spec.md [hunk 2]
```diff
@@ -42,6 +42,57 @@
...
+- [13. References](#13-references)
+- [About the ToIP Foundation](#about-the-toip-foundation)
+- [Appendix A: Consolidated Requirements](#appendix-a-consolidated-requirements)
+- [Appendix B: Consolidated Views of the ToIP Technology Stack](#appendix-b-consolidated-views-of-the-toip-technology-stack)
+    - [Functional Hourglass View](#functional-hourglass-view)
+    - [Sphere-of-Influence View](#sphere-of-influence-view)
+    - [Interaction Pattern View](#interaction-pattern-view)
+- [Appendix C: Mapping of Existing Technologies into the ToIP Technology Stack](#appendix-c-mapping-of-existing-technologies-into-the-toip-technology-stack)
+- [Footnotes](#footnotes)
+    - [^2](#2)
```
Comment (talltree) [inline spec.md]: Could we also remove this line (but leave the "Footnotes" line, as it's helpful to indicate there are footnotes).

**Review by talltree — CHANGES_REQUESTED** [review]
Great job! Two very minor changes requested.

File: spec.md (hunk 2 above)
Comment (andorsk) [reply spec.md]: yep sounds good

File: spec.md (hunk 1 above)
Comment (andorsk) [reply spec.md]: yep. sounds good

**Review by talltree — APPROVED** [review]
Fantastic job — in record time!!

**Comment by talltree** [comment]
LGTM. Merging as this completes all major tasks for PR1 except for the References section.

**Pull request merged**